main.go

```diff
@@ -107,26 +113,15 @@ func main() {
 	}
 }
 
-func registerNodeSelectorValidationWebhooks(mgr ctrl.Manager, skipper skipper.Skipper, conf Config) {
-	ans := &validate.AllowedLabels{}
-	for k, v := range conf.AllowedNodeSelectors {
-		if err := ans.Add(k, v); err != nil {
-			setupLog.Error(err, "unable to add allowed node selector")
-			os.Exit(1)
-		}
-	}
-
-	mgr.GetWebhookServer().Register("/validate-namespace-node-selector", &webhook.Admission{
-		Handler: &webhooks.NamespaceNodeSelectorValidator{
-			Skipper:               skipper,
-			AllowedNodeSelectors:  ans,
-			DenyEmptyNodeSelector: conf.NamespaceDenyEmptyNodeSelector,
-		},
-	})
-	mgr.GetWebhookServer().Register("/validate-workload-node-selector", &webhook.Admission{
-		Handler: &webhooks.WorkloadNodeSelectorValidator{
-			Skipper:              skipper,
-			AllowedNodeSelectors: ans,
+func registerNodeSelectorValidationWebhooks(mgr ctrl.Manager, conf Config) {
+	mgr.GetWebhookServer().Register("/mutate-pod-node-selector", &webhook.Admission{
+		Handler: &webhooks.PodNodeSelectorMutator{
+			Skipper: &skipper.NonOrganizationNamespaceSkipper{
```

Comment:
We might want to add the default selector to every pod with the nodeSelector `node-role.kubernetes.io/app: ""`? Not just the organization namespaces?

Reply:
Probably makes sense to do so, just to cover our bases. I think there's still a few of infra pods which aren't explicitly scheduled on the infra nodes, but I don't currently have a good idea if those pods require Plus nodes or if they'd be happy with Flex nodes on cloudscale.ch

Reply:
On second and third though i think this might be a bigger risk. Currently the webhook only applies to namespaces with the organisation label which is allows the cluster to create pods in system namespaces even if the agent is down.

https://github.com/appuio/component-appuio-cloud/blob/b0ffbd2f0efb9433ba40579dd98c1e558c50f070/tests/golden/defaults/appuio-cloud/appuio-cloud/01_agent/10_webhook_config.yaml#L18-L21

Reply:
I've removed the filter, so we could add a second webhook configuration in the component which applies to namespaces with the default label.